backup/restore.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+this_script="/opt/backup/restore.sh"
+
+set -euo pipefail
+
+backup_file="${1:-}"
+
+if [ ! -f "${backup_file}" ] ; then
+    echo "ERROR: missing backup tarball file."
+    echo "Usage: $0 <backup-tarball>"
+    echo "                  <backup-tarball> - backup created with 'backup.sh'"
+    echo
+    exit 1
+fi
+
+# Only sith lords and backup files deal in absolutes.
+backup_file="$(realpath "${backup_file}")"
+
+if    [[ "${backup_file}" == /opt/mastodon/* ]] \
+   || [[ "${backup_file}" == /opt/backup/* ]] \
+   || [[ "${backup_file}" == /opt/caddy/* ]] ; then
```

Comment:
I'm only wondering if it's possible that someone's going to have a wicked setup where `/opt/mastodon` is actually a symlink that points to exactly the directory where backup file resides, but these checks won't detect it. :)

And a single `=` is enough for checking equality.

Reply:
Interesting point, but then that would be a pretty thorough customisation, I would not expect our stock backup/restore to deal with that.

Reply:
Fair enough. And if `/opt` is a symlink then let's say that this is a self-inflicted pain.